docs/CONCEPTS.md

```diff
@@ -0,0 +1,124 @@
+
+> Golem connects computers in a peer-to-peer network, enabling both application
+> owners and individual users ("requestors") to rent resources of other users’
+> ("providers") machines. These resources can be used to complete tasks requiring any
+> amount of computation time and capacity.
+
+Since version `3.0`, `golem-js` leverages the notion of _renting compute resources_ and models the exposed domain APIs around it.
+
+### What it should do
+
+The primary tasks for the model to deliver are:
+
+- provide an abstraction over the complexity of _Golem Network Protocol_ and leverage the "rent compute resources, access agreement and cost information"
+- shorten the path of the user required to access a deployed instance of their workload
+- provide convenience APIs that allow easier exploration of the Golem Network domain (example: `rental.agreement.provider` allows _easier_ access to Provider information)
+
+### How it was done
+
+To deliver this vision, `golem-js` exposes the `ResourceRental` **aggregate** which wraps around the details of the _Golem Network Protocol_: the loosely coupled entities which it defines, and the processes/communication schemes which it requires. These include things _Agreements_, _Allocations_, _Activities_, _Invoices_, _DebitNotes_ and related conversations required by the protocol such as _debit note or invoice acceptance_ between Provider and Requestor (you).
+
+When using the `ResourceRental` model, you can still access these lower level domain objects via the APIs exposed by the `ResourceRental` object instance.
+
+### How is it different from TaskExecutor available in versions `1.x` and `2.x`
+
+`TaskExecutor` implemented the so-called _Task Model_, which is oriented around the notion of a _Task_ defined as a series of commands to be executed that have to succeed altogether for the task to be considered as successful. While this model is suitable for use-cases involving batch-map-reduce type of operations that can be distributed across many rented resources, it falls short in cases where users want to rent out resources and execute long-running activities such as hosting web services for many users (multi-tenancy).
+
+## Exe Unit
+
+### Why is it needed
+
+When you obtain resources via the Golem Network, your _Golem Virtual Machine Image (GVMI for short)_ is going to be deployed by `golem-js` into the _Activity_ running on the Provider. Technically, on the Provider instantiates an _ExeUnit_ which is a physical implementation of the _Activity_ of the Golem Network Protocol. Without digging in too much details, it's this _ExeUnit_ which at the end performs the operations that you issue via the Golem Network.
+
+If you're familiar with Docker, you can picture the architecture in the following way:
+
+| Golem              | Docker       | Kubernetes   |
+| ------------------ | ------------ | ------------ |
+| GVMI               | Docker Image | Docker Image |
+| Activity (ExeUnit) | Container    | Container    |
```

Comment:
why even show k8s here if it's the same as Docker?

Reply:
I'm dropping this comparision as it's not helpful at all and after talking with Witek from the Core Team I identified too many holes in this comparision.